packages/react-dom/src/client/focus/__tests__/TabFocusContainer-test.internal.js

```diff
@@ -0,0 +1,242 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+let React;
+let ReactFeatureFlags;
+let TabFocusContainer;
+
+const createTabForward = type => {
+  const event = new KeyboardEvent('keydown', {
+    key: 'Tab',
+    bubbles: true,
+    cancelable: true,
+  });
+  return event;
+};
+
+const createTabBackward = type => {
+  const event = new KeyboardEvent('keydown', {
+    key: 'Tab',
+    shiftKey: true,
+    bubbles: true,
+    cancelable: true,
+  });
+  return event;
+};
+
+describe('TabFocusContainer', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    ReactFeatureFlags = require('shared/ReactFeatureFlags');
+    ReactFeatureFlags.enableScopeAPI = true;
+    ReactFeatureFlags.enableFlareAPI = true;
+    TabFocusContainer = require('../TabFocusContainer').TabFocusContainer;
+    React = require('react');
+  });
+
+  describe('ReactDOM', () => {
+    let ReactDOM;
+    let container;
+
+    beforeEach(() => {
+      ReactDOM = require('react-dom');
+      container = document.createElement('div');
+      document.body.appendChild(container);
+    });
+
+    afterEach(() => {
+      document.body.removeChild(container);
+      container = null;
+    });
+
+    it('should work as expected with simple tab operations', () => {
+      const inputRef = React.createRef();
+      const input2Ref = React.createRef();
+      const buttonRef = React.createRef();
+      const butto2nRef = React.createRef();
+      const divRef = React.createRef();
+
+      const Test = () => (
+        <TabFocusContainer>
+          <input ref={inputRef} />
+          <button ref={buttonRef} />
+          <div ref={divRef} tabIndex={0} />
+          <input ref={input2Ref} tabIndex={-1} />
+          <button ref={butto2nRef} />
+        </TabFocusContainer>
+      );
+
+      ReactDOM.render(<Test />, container);
+      inputRef.current.focus();
+      document.activeElement.dispatchEvent(createTabForward());
```

Comment:
You could use the testing-library helpers for this

Reply:
Sure. This was mostly a copy and paste from the old focus scope tests